src/app/accounts/create-account.ts

```diff
@@ -89,30 +87,31 @@ export const createUserForPhoneSchema = async ({
   const user = await UsersRepository().persistNew(userRaw)
   if (user instanceof Error) return user
 
-  const account = await setupAccount({ userId: user.id, config, phoneNumberValid })
+  let account = await AccountsRepository().findByUserId(user.id)
+  if (account instanceof Error) return account
+
+  account = await setupAccount({ account, config, phoneNumberValid })
   if (account instanceof Error) return account
 
-  return user
+  return account
 }
 
 // kratos user already exist, as he has been using self registration
-export const createUserForEmailSchema = async ({
+export const createAccountForEmailSchema = async ({
   kratosUserId,
   config,
 }: {
   kratosUserId: string
   config: AccountsConfig
-}) => {
+}): Promise<Account | RepositoryError> => {
   const kratosUserIdValid = checkedToKratosUserId(kratosUserId)
   if (kratosUserIdValid instanceof Error) return kratosUserIdValid
 
-  const user = await UsersRepository().persistNewKratosUser({
-    kratosUserId: kratosUserIdValid,
-  })
-  if (user instanceof Error) return user
+  let account = await AccountsRepository().persistNewKratosUser(kratosUserIdValid)
```

Comment:
We're probably approaching the point where we'd need to separate Accounts from Users at the persistence layer as well.

```suggestion
  let account = await AccountsRepository().persistNewByKratosUserId(kratosUserIdValid)
```